### PR TITLE
Add Note for installing test group gems.

### PR DESCRIPTION
**Pull request by joshuapinter**
Added a note that in order to run the specs, you may need to install the bundle with the `--with test` command line option in order to install the `test` group of gems, including `rake`. Otherwise you end up with an error like this: 

```bash
$ be rake spec

bundler: failed to load command: rake (/Users/joshuapinter/.rbenv/versions/2.6.5/bin/rake)
Gem::Exception: can't find executable rake for gem rake. rake is not currently included in the bundle, perhaps you meant to add it to your Gemfile?
```

Wasted 15 minutes figuring out what the issue was so this should help others contribute.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ Thank you!
     ```
     bundle exec rake spec
     ```
+    
+    _NOTE: You may need to install the bundle with the `test` group first by running `bundle install --with test`. Otherwise, you will get `can't find executable rake for gem rake` error._
```
Comment (imjoehaines) [inline CONTRIBUTING.md:23]: I think it makes sense to have the installation step as its own bullet point in the contributing guide as hopefully then contributors won't see the error at all. So maybe something like this as the first bullet in this "Polish" section would be better:

```markdown
* Install the test dependencies
    ```
    bundle install --with test sidekiq --binstubs
    ```
```

**Review by imjoehaines — CHANGES_REQUESTED** [review]
Hey @joshuapinter, thanks for the PR!

There is a bundle install command in [TESTING.md](https://github.com/bugsnag/bugsnag-ruby/blob/master/TESTING.md) already, but it's not clear enough that it's a required setup step (it looks like part of the command to run the tests instead), so I think this change makes sense

As it's required before the command to run the tests will work, I think it makes sense to be its own bullet point, especially as there shouldn't be any harm in re-running it

I'll have a look at updating the testing guide to be clearer about the setup as well

File: CONTRIBUTING.md (hunk 1 above)
Comment (joshuapinter) [reply CONTRIBUTING.md:23]: Also, in the README it says the command to run the tests is `bundle exec rake spec` whereas the TESTING file says `bundle exec rake`. Which one do you prefer?

**Comment by joshuapinter** [comment]
K, updated to a separate bullet point for installing test bundle.

File: CONTRIBUTING.md (hunk 1 above)
Comment (imjoehaines) [reply CONTRIBUTING.md]: I think following the testing guide makes sense, so `bundle exec rake` — `spec` is the default so there's no real difference 🙂

TESTING.md isn't mentioned in the readme, but I think that's OK after this PR. If all the required steps for the unit tests are in CONTRIBUTING (which is mentioned in the readme) then the rest of the testing guide is irrelevant for outside contributors as the end to end tests can only be run with private credentials

**Comment by joshuapinter** [comment]
Also changed `bundle exec rake spec` to just `bundle exec rake` for consistency with `TESTING.md`.

**Review by imjoehaines — APPROVED** [review]
Thanks @joshuapinter!

**Comment by johnkiely1** [comment]
This has now been added in v6.18.0.

**Comment by joshuapinter** [comment]
👍

**Pull request merged**